File: genetic-car-starters/genetic-car-starter-csharp/GeneticAlgorithm/Api/CarView.cs
using System.Runtime.Serialization;

namespace GeneticAlgorithm.Api
{
    [DataContract]
    public class CarView
    {
        [DataMember(Name="chassis")]
        public readonly Chassis Chassis = new Chassis();

        [DataMember(Name="wheel1")]
        public readonly Wheel Wheel1 = new Wheel();

        [DataMember(Name="wheel2")]
        public readonly Wheel Wheel2 = new Wheel();

        public override string ToString()
        {
            return "CarView{" +
                    "chassis=" + Chassis +
                    ", wheel1=" + Wheel1 +
                    ", wheel2=" + Wheel2 +
                    '}';
        }
    }
}

File: genetic-car-starters/genetic-car-starter-java/src/main/java/fr/genetic/client/java/api/CarView.java
package fr.genetic.client.java.api;

import java.util.ArrayList;
import java.util.List;

public class CarView {

    public Chassis chassis = new Chassis();
    public Wheel wheel1 = new Wheel();
    public Wheel wheel2 = new Wheel();

    public static class Chassis {
        public List<Float> vecteurs = new ArrayList<>();
        public float densite;

        @Override
        public String toString() {
            return "Chassis{" +
                    "vecteurs=" + vecteurs +
                    ", densite=" + densite +
                    '}';
        }
    }

    public static class Wheel {
        public float radius;
        public float density;
        public int vertex;

        @Override
        public String toString() {
            return "Wheel{" +
                    "radius=" + radius +
                    ", density=" + density +
                    ", vertex=" + vertex +
                    '}';
        }
    }

    @Override
    public String toString() {
        return "CarView{" +
                "chassis=" + chassis +
                ", wheel1=" + wheel1 +
                ", wheel2=" + wheel2 +
                '}';
    }
}
